src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -292,6 +293,17 @@
         optional=True,
     ),
     TemporaryTransientGrammar=OneOf(Ref("TemporaryGrammar"), "TRANSIENT"),
+    BaseExpressionElementGrammar=OneOf(
+        # Allow use of CONNECT_BY_ROOT pseudo-columns.
+        # https://docs.snowflake.com/en/sql-reference/constructs/connect-by.html#:~:text=Snowflake%20supports%20the%20CONNECT_BY_ROOT,the%20Examples%20section%20below.
+        Sequence("CONNECT_BY_ROOT", Ref("ColumnReferenceSegment")),
```

Comment:
N.B. the level_expression pseudo-columns are pretty poorly documented in the docs.
As far as I can see there is:
- `level` which just acts as a standard `ColumnReferenceSegment`
- `connect_by_root` which has an example in the docs and is handled here
- `connect_by_path` mentioned in [this](https://docs.snowflake.com/en/user-guide/queries-hierarchical.html#:~:text=The%20CONNECT%20BY%20syntax%20supports%20convenient%20pseudo%2Dcolumns%20such%20as%20LEVEL%2C%20CONNECT_BY_ROOT%2C%20and%20CONNECT_BY_PATH) doc section. However it does not seem to work the same way as either `level` or `connect_by_root` and there's zero explanation in the docs so I've omitted it from this PR. Users will likely use the alternative `SYS_CONNECT_BY_PATH` function instead (which we already parse) since that has an actual example.